0000-product.md

```diff
@@ -0,0 +1,488 @@
+- Feature Name: Product
+- Start Date: 5/13/19
+- RFC PR: [#5](https://github.com/hyperledger/grid-rfcs/pull/5)
+
+# Summary
+[summary]: #summary
+
+
+This RFC proposes a generic and extensible framework for a Hyperledger Grid
+Product entity as well as a more specific - and fully encapsulated - GS1
+compliant Product.
+
+GS1 is a widely used data standard in enterprises and, given that
+positioning and familiarity, Grid support feels natural.  Additional
+implementations of Product for specialized industries or use cases may derive
+from or extend this implementation.
+
+_For the base implementation of a product
+on grid, there will be 4 fields. A product_id, a product_type, an owner, and
+properties (a repeated key-value field)._
+
+
+# Motivation
+[motivation]: #motivation
+
+The Grid Product implementation is designed for sharing product master data
+between participants. Product is a near universal concept within supply chains
+solutions and would naturally be one of the highest areas of re-use across Grid
+applications.
+
+The design will address use cases including:
+
+- Sharing of Product master data across a network
+- Including Product in other business transactions (track and trace events,
+purchases, sales, etc)
+- Enriching UX experiences by including additional attribution of a Product
+such as names or descriptions
+
+It is also useful to use product information as
+auxiliary data in other supply chain solutions.  For example, in track and
+trace, the location and temperature of an item (an instance of a product) is
+stored.  When presenting this information later to a human user, it is useful
+to combine it with information about the product, such as the product’s name
+and description.  Therefore, Product becomes a reusable and important component
+within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_type__**. This
```

Comment:
I'm still not sure I like the notion of `Product Type` in this context, as that is a heavily overloaded term when talking about Products.  I'm struggling to come up with a better term to use here, though.  Something like "Product Schema" or something, maybe?

Reply:
Perhaps product_standard?

Reply:
per discussions in RC, `product_namespace` is the agreed upon term.

Reply:
Updated